Tasks/RunDistributedTests/RunDistributedTests.ps1

```diff
@@ -24,6 +24,95 @@ Function CmdletHasMember($memberName) {
     return $cmdletParameter
 }
 
+#To parse the Overriding Parameters string provided by user
+Function GetPropertiesMapping {
+    param ( [String] $overridingParameters )
+    
+        $inputStr = $overridingParameters
+        $parameterStrings = New-Object System.Collections.Generic.List[String]
+        $startOfSegment = 0;
+        $index = 0;
+    
+        $separator = ";" 
+        $escapeCharacter = '\'
+    
+        while ($index -ilt $inputStr.Length)
+        {
+            $index = $inputStr.indexOf($separator, $index);
+            if ($index -gt 0 -and $inputStr[$index - 1] -eq $escapeCharacter)
+            {
+                $index += $separator.Length;
+                continue; 
+            }
+            if ($index -eq -1)
+            {
+                break;
+            }
+            $parameterStrings.Add($inputStr.Substring($startOfSegment, $index - $startOfSegment).Replace($escapeCharacter + $separator, $separator));
+            $index += $separator.Length;
+            $startOfSegment = $index;
+        }
+        $parameterStrings.Add($inputStr.Substring($startOfSegment).Replace($escapeCharacter + $separator, $separator));
+        
+        $properties = New-Object 'system.collections.generic.dictionary[string,string]'
+        foreach ($s in $parameterStrings)
+        {
+            $pair = $s.Split('=', 2);
+            if ($pair.Length -eq 2)
+            {
+               if (!$properties.ContainsKey($pair[0]))
+               {
+                    $properties.Add($pair[0], $pair[1])
+               }
+            }
+        }
+        return $properties;
+    }
+    
+Function OverrideTestSettingProperties {
+    param (
+    [String] $overridingParameters, 
+    [System.Xml.XmlDocument] $testSettings
+    )
+    
+        if([String]::IsNullOrEmpty($overridingParameters) -or $testSettings -eq $null)
+        {
+            return $false
+        }
+    
+        $properties = GetPropertiesMapping -overridingParameters $overridingParameters
+        $propertyNodes = $testSettings.TestSettings.Properties.ChildNodes
+    
+        $hasOverridenProperties  = $false;
+        foreach($property in $propertyNodes)
+        {       
+            if ([string]::CompareOrdinal($property.LocalName, "Property") -ne 0 ) { continue } 
+    
+            $nameAttribute = $property.name
+            $valueAttribute = $property.value
+    
+            if(-not $nameAttribute) { $nameAttribute = $property.Name }
+    
+            if(-not $valueAttribute) { $valueAttribute = $property.Value }
+            
+            if (-not ($nameAttribute -and $valueAttribute)) { continue }
+             
```

Comment:
indention

Reply:
rectified

---
In reply to: [137520931](https://github.com/Microsoft/vsts-tasks/pull/5246#discussion_r137520931) [](ancestors = 137520931)